### PR TITLE
More debugging/trying to find the error:

### PR DESCRIPTION
**Pull request by nikkolasg**
Created the PR so we can discuss about it

### CODE REVIEW

File: lib/network/net.go [hunk 1]
```diff
@@ -185,15 +200,24 @@ func (c *TcpConn) Send(ctx context.Context, obj ProtocolMessage) error {
 	}
 	//c.Conn.SetWriteDeadline(time.Now().Add(timeOut))
 	// First write the size
-	packetSize := Size(len(b))
-	if err := binary.Write(c.Conn, globalOrder, packetSize); err != nil {
+	packetSize := uint64(len(b))
+
+	// FIXME Write the size as an UvarInt
+	var sb = make([]byte, binary.MaxVarintLen64)
+	n := binary.PutUvarint(sb[:], packetSize)
```
Comment (nikkolasg) [inline lib/network/net.go]: That's not necessary, it does not change anything. if you look at the source of binary.Write ([code](https://golang.org/src/encoding/binary/binary.go?s=6397:6461#L224), it uses PutUVarint as the underlying function when you give it a uint64.

Comment (Liamsi) [inline lib/network/net.go]: Yepp, saw that. It's even more strange, that it again behaves differently from before.

Comment (Liamsi) [inline lib/network/net.go]: I did `git checkout sda_1 -- net.go`  and the same with struct.go and encoding.go to see if messed up sth: As expected it does not block but I now get a `networfatal error: bad map state` [sic] and `invalid memory address or nil pointer dereference` and much more. So at least our changes didn't make it worse...

Comment (Liamsi) [inline lib/network/net.go]: > That's not necessary, it does not change anything. if you look at the source of binary.Write (code, it uses PutUVarint as the underlying function when you give it a uint64.
See the next commit

**Comment by ineiti** [comment]
Is there a failing test-case? Ways to reproduce? Why is this in 'Ready4Merge'?

**Comment by Liamsi** [comment]
> Is there a failing test-case? Ways to reproduce? Why is this in 'Ready4Merge'?

Run the pbft simulation with ~90 hosts and blocksize > 2000 on deterlab OR with even larger blocksize (and fewer machines) on localhost

**Comment by Liamsi** [comment]
> Why is this in 'Ready4Merge'?

Only because it's the default stage. Nicolas and just needed a PR so that we can continue discussing on code changes/experiments during the weekend.

**Comment by Liamsi** [comment]
Closed as this PR lead to some confusion in a sandstorm... The error persists though: using a broadcast protocol (ignores the tree) sending around large pieces of data (some MB), the Receive function cannot read the package size correctly. In worst case this means it reads more than it should (we read some blockchain data/preprepare  message together with other small messages sent later ). Before the package size received was even negative (smelled like an integer overflow), with our changes it now is just too big.this only happens with a larger number of hosts. It took us days and adding a lot of debug output (in all parts of the code, even in protobuf) to figure that out. 
I will open a new issue with a more precise description if we decide to postpone working in this bug.

**Pull request closed without merging**